### PR TITLE
lib: bin: lwm2m_carrier: update to v3.2.0

### PR DESCRIPTION
**Pull request by nordic-hani**
- Updated library to the latest build.
- Responsibility for initializing and shutting down the modem library has been moved to the application.
- See the changelog in this PR (doc/nrf/libraries/bin/lwm2m_carrier/CHANGELOG.rst) or the generated documentation to see detailed changes.

Signed-off-by: Håvard Vermeer [havard.vermeer@nordicsemi.no](mailto:havard.vermeer@nordicsemi.no)

### CODE REVIEW

**Review by SeppoTakalo — APPROVED** [review]

**Review by simensrostad — APPROVED** [review]
ATv2 changes looks good

**Review by tokangas — APPROVED** [review]
`modem_shell` changes look good.

**Review by VTPeltoketo — APPROVED** [review]
Tested against SLM, looks good

**Comment by kacperradoszewski** [comment]
@simensrostad please have a look at the ATv2 changes again, the LTE LC initialization was moved to the modem init hook, is this ok to do for all the modules or should we restrict that to carrierlib only?

**Review by kacperradoszewski — APPROVED** [review]

**Review by stig-bjorlykke — APPROVED** [review]

**Review by nordic-mare — APPROVED** [review]

File: applications/asset_tracker_v2/sample.yaml [hunk 1]
```diff
@@ -125,6 +125,7 @@ tests:
     integration_platforms:
       - nrf9160dk_nrf9160_ns
     extra_args: OVERLAY_CONFIG=overlay-carrier.conf
+                DTC_OVERLAY_FILE=nrf9160dk_with_ext_flash.overlay
```
[the remark this answers is not in the document]
Comment (nordic-mare) [reply applications/asset_tracker_v2/sample.yaml:128]: It is needed to make it fit in CI.

**Review by umapraseeda — APPROVED** [review]

**Pull request merged**